File: Data Structures & Algorithms/is-anagram/submission-1.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;

        vector<int> freq_s(26, 0);
        vector<int> freq_t(26, 0);

        for (char c : s) {
            int index = c - 'a';
            freq_s[index]++;
        }
        for (char c : t) {
            int index = c - 'a';
            freq_t[index]++;
        }

        for (int i=0; i<26; i++) {
            if (freq_s[i] != freq_t[i]) return false;
        }
        return true;
    }
};

File: Data Structures & Algorithms/count-connected-components/submission-1.py
class Solution:
    def countComponents(self, n: int, edges: List[List[int]]) -> int:
        adjacency_list = defaultdict(list)
        for edge in edges:
            adjacency_list[edge[0]].append(edge[1])
            adjacency_list[edge[1]].append(edge[0])

        print(adjacency_list)
        visited = [False]*n

        def dfs(u):
            visited[u] = True
            for v in adjacency_list[u]:
                if not visited[v]:
                    dfs(v)

        counter = 0
        while not all(visited):
            counter +=1
            for index, val in enumerate(visited):
                if not val:
                    dfs(index)
                    break

        return counter